pkg/steps/source.go

```diff
@@ -129,27 +139,33 @@ func (s *sourceStep) Run(ctx context.Context, dry bool) error {
 		return fmt.Errorf("could not resolve clonerefs source: %v", err)
 	}
 
-	return handleBuild(s.buildClient, createBuild(s.config, s.jobSpec, clonerefsRef, s.resources, s.sshSecretName), dry, s.artifactDir, s.dryLogger)
+	return handleBuild(s.buildClient, createBuild(s.config, s.jobSpec, clonerefsRef, s.resources, s.sshSecretName, s.oauthSecretName), dry, s.artifactDir, s.dryLogger)
+}
+
+func determineCloneURI(sshSecretName, oauthSecretName, org, repo string) string {
+	if len(sshSecretName) > 0 {
+		return fmt.Sprintf("ssh://git@github.com/%s/%s.git", org, repo)
+	}
+	if len(oauthSecretName) > 0 {
+		return fmt.Sprintf("https://github.com/%s/%s.git", org, repo)
+	}
+	return ""
```

Comment:
This is not a valid URI and will cause a crash if we allow it

Reply:
The value is already empty if it reaches that point.

Reply:
That was not the behavior before and we should not allow it.

Reply:
I wonder why the tests passed. I'll check.